Classes/common/query/CDTQIndexUpdater.m

```diff
@@ -397,12 +396,14 @@ + (CDTQSqlParts *)createPartsForFieldNames:(NSArray *)fieldNames
             [CDTQValueExtractor extractValueForFieldName:fieldName fromDictionary:rev.body];
 
         if (value) {
+            // Do not include an empty array field as you would not include a field with no value.
+            if ([value isKindOfClass:[NSArray class]] && ((NSArray *)value).count == 0) {
+                continue;
+            }
+            
             [includedFieldNames addObject:fieldName];
             [args addObject:value];
```

Comment:
Same answer as sync-android PR.  Take a look a few lines above at https://github.com/cloudant/CDTDatastore/pull/165/files#diff-5440a5114825a38f7d2feeefa446aabfR391.  This will skip fields and arguments included with initial settings.